### PR TITLE
Changed traversion of the messages...

### PR DESCRIPTION
**Pull request by sauin**
… during blob restoration if snapshot was deleted

### CODE REVIEW

File: v1_x/git-remote-gosh/src/git_helper/fetch/mod.rs [hunk 1]
```diff
@@ -204,16 +213,22 @@ where
...
+                        let branch: GetNameBranchResult = parent_contract.run_static(
+                            self.blockchain.client(),
+                            "getNameBranch",
+                            None
+                        ).await?;
+                        tracing::debug!("commit={id}: extracted branch {:?}", branch.name);
+                        branches.insert(branch.name);
+                    }
+                } else {
+                    branches.insert(branch.to_owned());
```
Comment (SilkovAlexander) [inline v1_x/git-remote-gosh/src/git_helper/fetch/mod.rs:231]: Isn't it the same as ^

File: v1_x/git-remote-gosh/src/git_helper/fetch/mod.rs [hunk 2]
```diff
@@ -204,16 +213,22 @@ where
...
-                for parent in onchain_commit.parents {
-                    let parent = BlockchainContractAddress::new(parent);
-                    let parent_contract = GoshContract::new(&parent, gosh_abi::COMMIT);
-                    let branch: GetNameBranchResult = parent_contract.run_static(
-                        self.blockchain.client(),
-                        "getNameBranch",
-                        None
-                    ).await?;
-                    branches.insert(branch.name);
+                branches.insert(onchain_commit.branch.clone());
```
Comment (SilkovAlexander) [inline v1_x/git-remote-gosh/src/git_helper/fetch/mod.rs:216]: ->

File: v3_x/git-remote-gosh/src/blockchain/snapshot/load/diffs/iterator.rs [hunk 3]
```diff
@@ -133,6 +133,10 @@ impl DiffMessagesIterator {
                     &file_path,
                 )
                 .await?;
+                let snapshot_contract =
+                    GoshContract::new(original_snapshot.clone(), crate::abi::SNAPSHOT);
+                let snapshot_is_active = snapshot_contract.is_active(client).await?;
+                tracing::trace!("snap={original_snapshot} is {}", if snapshot_is_active { "ACTIVE" } else { "NON_ACTIVE" });
```
Comment (SilkovAlexander) [inline v3_x/git-remote-gosh/src/blockchain/snapshot/load/diffs/iterator.rs:139]: v3 version of this file has more logs than others, I suggest add this extra tracing strings to at least the last version

**Review by andrew-kurochkin — CHANGES_REQUESTED** [review]
Please-please add comments to tests

**Review by SilkovAlexander — DISMISSED** [review]

**Review by andrew-kurochkin — APPROVED** [review]

**Pull request merged**